### DISCUSSION TITLE
Build venv from lock file

### DISCUSSION THREAD

**Discussion by vanyasvl** [astral-sh/rye#627] [General]
I can't find how to build venv from a lock file. rye sync creates venv from the pyproject.toml and generates lock file. But how to create venv with rye from the created lock file. 
We want to use rye to create reproducible venvs from lockfile (virtual project) instead of poetry.

**Comment by bluss** [comment]
Does `rye sync --no-lock` help you? It is sync but without updating the lockfile, instead using the lockfile as it is.

**Reply by vanyasvl** [reply]
No, `rye sync --no-lock` doesn't use lock file for dependencies, it uses pyproject.toml and latest version of dependencies and it's not reproducible. 
For example `poetry install` generates venv from the lock file, maybe you can add virtual project support to the `rye install`?

**Reply by bluss** [reply]
Does it fail in general or is this some corner case? I don't quite follow you, since rye sync --no-lock requires that the lockfiles are present and it uses the versions it says, if I edit it manually.
